core/src/main/scala/io/finch/Finch.scala

```diff
@@ -0,0 +1,16 @@
+package io.finch
+
+import cats.effect.Effect
+import io.finch.endpoint.effect.EffectEndpoints
+import io.finch.syntax.effect.EffectEndpointMappers
+
+abstract class Finch[F[_]](implicit val E: Effect[F])extends EffectEndpoints[F]
```

Comment:
How about this name it `Module[F]`? I think that's what iteratee is doing.

I'd imagine something like this should work:

```scala
object Main extends App with Module[IO] {
  def foo = param("a")
}
```